hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/transform/ProcessorTransform.java

```diff
@@ -89,12 +94,23 @@ public static ProcessorTransform customProcessorTransform(
         //      be sent to a random member. We keep it this way for simplicity:
         //      the number of in-flight items is limited (maxAsyncOps)
         return new ProcessorTransform(operationName + "UsingContextAsync", upstream,
-                ProcessorMetaSupplier.of(flatMapUsingContextAsyncP(contextFactory, Object::hashCode, flatMapAsyncFn)));
+                ProcessorMetaSupplier.of(flatMapUsingContextAsyncP(contextFactory, Object::hashCode, flatMapAsyncFn)),
+                contextFactory.isCooperative());
     }
 
     @Override
     public void addToDag(Planner p) {
-        PlannerVertex pv = p.addVertex(this, name(), localParallelism(), processorSupplier);
+        PlannerVertex pv = p.addVertex(this, name(), determineLocalParallelism(), processorSupplier);
         p.addEdges(this, pv.v);
     }
+
+    protected int determineLocalParallelism() {
```

Comment:
Rather than this it would be better to use ProcessorMetaSupplier with a preferred parallelism. The difference is that if, for some reason, there's only 1 cpu core, it will use 1 thread instead of 2.